reo/src/data_manager.py

```diff
@@ -972,13 +1021,17 @@ def _get_time_steps_with_grid(self):
             time_steps_without_grid -- list of ints indicating outage time
                 steps
         """
-        time_steps_with_grid = list()
-        time_steps_without_grid = list()
-        for i, pf in enumerate(self.util.prod_factor):
-            if pf > 0.5:
-                time_steps_with_grid.append(i+1)
-            else:
-                time_steps_without_grid.append(i+1)
+        if self.off_grid_flag:
+            time_steps_with_grid = []
+            time_steps_without_grid = list(range(8760)[1:])
```

Comment:
@adfarth do you know why `time_steps_without_grid` is a range from 1 to 8759 and not 1 to 8760? 

Also, are offgrid analyses limited to an hourly time resolution?

Reply:
@NLaws It looks like currently in [validators.py](https://github.com/NREL/REopt_Lite_API/blob/ea9edd6d5215325fa0373ecfef219f00040033d5/reo/validators.py#L499) there is code to allow for multiple time steps per hour: 

`            if self.off_grid_flag:
                self.input_dict["Scenario"]["Site"]["LoadProfile"]["outage_start_time_step"] = 1
                # Previously was not modeling tax incentives in offgrid scenarios 
                # self.input_dict["Scenario"]["Site"]["Financial"]["offtaker_tax_pct"] = 0.0
                # self.input_dict["Scenario"]["Site"]["Financial"]["owner_tax_pct"] = 0.0
                if self.input_dict["Scenario"]["time_steps_per_hour"] == 4:
                    self.input_dict["Scenario"]["Site"]["LoadProfile"]["outage_end_time_step"] = 35040
                elif self.input_dict["Scenario"]["time_steps_per_hour"] == 2:
                    self.input_dict["Scenario"]["Site"]["LoadProfile"]["outage_end_time_step"] = 17520
                else:
                    self.input_dict["Scenario"]["Site"]["LoadProfile"]["outage_end_time_step"] = 8760`

But the code does break when trying to run multiple time steps per hour (likely related to the range you've referenced in data_manager.py. I will look into this and propose a solution.

Reply:
@NLaws  I think it makes sense to entirely remove the `if self.off_grid_flag and just use the else: code as below: `
```suggestion
                    time_steps_with_grid = list()
        time_steps_without_grid = list()
        for i, pf in enumerate(self.util.prod_factor):
            if pf > 0.5:
                time_steps_with_grid.append(i+1)
            else:
                time_steps_without_grid.append(i+1)
```


I tested this and it seems to be working, and can push the change.